app/src/org/commcare/suite/model/graph/XYSeries.java

```diff
@@ -76,23 +76,35 @@ public void setY(String y) {
         mYParse = null;
     }
 
+    public void removeConfiguration(String key) {
+        mConfiguration.remove(key);
+    }
+
     @Override
     public void setConfiguration(String key, Text value) {
-        mConfiguration.put(key, value);
+        if (mPointConfiguration.keySet().contains(key)) {
+            mPointConfiguration.remove(key);
```

Comment:
Minor, but I think this is redundant; the `put` below will overwrite the current value.